diff --git a/imap/httpd.c b/imap/httpd.c
index dc53f8c..24b65e5 100644
--- a/imap/httpd.c
+++ b/imap/httpd.c
@@ -2202,7 +2202,7 @@ EXPORTED time_t calc_compile_time(const char *time, const char *date)
     memset(&tm, 0, sizeof(struct tm));
     tm.tm_isdst = -1;
     sscanf(time, "%02d:%02d:%02d", &tm.tm_hour, &tm.tm_min, &tm.tm_sec);
-    sscanf(date, "%s %2d %4d", month, &tm.tm_mday, &tm.tm_year);
+    sscanf(date, "%3s %2d %4d", month, &tm.tm_mday, &tm.tm_year);
     tm.tm_year -= 1900;
     for (tm.tm_mon = 0; tm.tm_mon < 12; tm.tm_mon++) {
         if (!strcmp(month, monthname[tm.tm_mon])) break;
diff --git a/imap/ical_support.c b/imap/ical_support.c
index 1d7550a..e1bda50 100644
--- a/imap/ical_support.c
+++ b/imap/ical_support.c
@@ -458,7 +458,7 @@ const char *get_icalcomponent_errstr(icalcomponent *ical)
 
                 /* Check if this is an empty property error */
                 if (sscanf(errstr,
-                           "No value for %s property", propname) == 1) {
+                           "No value for %255s property", propname) == 1) {
                     /* Empty LOCATION is OK */
                     if (!strcasecmp(propname, "LOCATION")) continue;
                     if (!strcasecmp(propname, "COMMENT")) continue;
